src/shared/backend/services/cloudbuild.ts

```diff
@@ -88,6 +88,37 @@ export const createJob = async (params: JobStatusParams): Promise<Job> => {
   return data as Job;
 };
 
+// Query a job, if it's not found, create it
+export const queryCreateJob = async (params: JobStatusParams): Promise<Job> => {
+  try {
+    return await queryJobStatus(params);
+  } catch (error) {
+    return await createJob(params);
+  }
+};
+
+function timeout(ms: number): Promise<void> {
+  /* eslint-disable-next-line no-promise-executor-return */
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+export const queryJobStatusTillSucess = async (
```

Comment:
Nice. Though it could be `waitForJobSuccess`. Maybe we could throw an error if the job fails and catch that in the flash job function?